framework/web/Request.php

```diff
@@ -1477,6 +1493,181 @@ protected function loadCookies()
         return $cookies;
     }
 
+    /**
+     * Returns uploaded files for this request.
+     * Uploaded files are returned in format according to [PSR-7 Uploaded Files specs](http://www.php-fig.org/psr/psr-7/#16-uploaded-files).
+     * @return array uploaded files.
+     * @since 2.1.0
+     */
+    public function getUploadedFiles()
+    {
+        if ($this->_uploadedFiles === null) {
+            $this->getBodyParams(); // uploaded files are the part of the body and may be set while its parsing
+            if ($this->_uploadedFiles === null) {
+                $this->_uploadedFiles = $this->defaultUploadedFiles();
+            }
+        }
+        return $this->_uploadedFiles;
+    }
+
+    /**
+     * Sets uploaded files for this request.
+     * Data structure for the uploaded files should follow [PSR-7 Uploaded Files specs](http://www.php-fig.org/psr/psr-7/#16-uploaded-files).
+     * @param array|null $uploadedFiles uploaded files.
```

Comment:
Could the param type be specified as  `Psr\Http\Message\UploadedFileInterface[]|null`?

Reply:
Same as https://github.com/yiisoft/yii2/pull/14801#discussion_r138607659